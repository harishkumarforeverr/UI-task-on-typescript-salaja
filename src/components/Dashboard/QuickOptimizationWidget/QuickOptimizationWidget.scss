.QuickOptimizationWidget {
  padding-left: 2%; 
  margin-top: 20px;
  margin-left: 20px;
  margin-bottom: 20px;
  padding-right: 20px;
  height: 100%;
  padding-right: 3%;
  .Conatiner2sIconsTitle {
    height: 100%;
    .QuickOptimizationWidget_hardwareSetup {
      display: flex;
      gap: 2rem;
      .hardwareSetup {
        display: flex;
        gap: 2rem;
        min-width: 887px;
        min-height: 373px;
        padding-left: 1rem;
        background: #f2f8fb;
        border: 1px solid #118899;
        border-radius: 12px;
        .hardWareTitleAndImage {
          h1 {
            font-family: "Helvetica";
            font-style: normal;
            font-weight: 400;
            font-size: 22px;
            line-height: 24px;
            

            color: #118899;
          }
          img {
            width: 378.73px;
            height: 300px;
          }
        }
        .hardWareDescription {
          h1 {
            font-family: "Helvetica";
            font-style: normal;
            font-weight: 700;
            font-size: 14px;
            line-height: 18px;
            color: #118899;
          }
          ol {
            li {
              font-family: Roboto;
              font-style: normal;
              font-weight: 400;
              font-size: 14px;
              line-height: 18px;
              color: #231f20;
              margin-bottom: 7px;
            }
          }
          p {
            padding-right: 4%;
            margin-left: 3rem;
            font-family: Roboto;
            font-style: normal;
            font-weight: 400;
            font-size: 14px;
            word-break: break-all;
            line-height: 20px;
            line-height: 1.6rem; 
            color: #231f20;
          }
          .setupdone {
            margin-left: 3rem;
            margin-top: 1.5rem;
            h1 {
              font-family: "Helvetica";
              font-style: normal;
              font-weight: 700;
              font-size: 18px;
              line-height: 24px;
              

              color: #118899;
            }
            .icon {
              .anticon {
                font-size: 1.3rem;
              }
              margin-right: 10px;
              
            }
          }
          .actionBtn {
            margin-left: 3rem;
            display: flex;
            align-items: baseline;
            gap: 1rem;
            margin-top: 3rem;
            p {
              white-space: nowrap;
              margin: 0;
              font-family: Helvetica;
              font-style: normal;
              font-weight: 700;
              font-size: 14px;
              line-height: 24px;
            
              color: #118899;
            }
            button {
              font-family: "Helvetica";
              font-style: normal;
              font-weight: 400;
              font-size: 14px;
              line-height: 24px;
              border: 2px solid #cc0000;
              color: #cc0000;
              height: 33px;
              min-width: 6.5rem;
              margin-left: auto;
              margin-right: 5rem;
              border-radius: 0px;
            }
            .Setup_completed_btn{
              background-color: #cc0000;;
              color: white;
              margin-right: 1rem;
              border-radius: 0px;
            }
          }
        }
      }
    }
    .cardsConatiner {
      display: flex;
    
      justify-content: space-between;
      //  edi gap
      gap: 5%;
      margin-top: 2rem;
      margin-left: 1rem;
      .cardsConatiner_card {
        //  edi width
        flex-basis: 100%;
        border-bottom-right-radius: 2%;
        overflow: hidden;
      }
      .cards {
        height: 100%;
        padding-top: 1rem;

       
        text-align: center;
        
        position: relative;
        box-sizing: border-box;
         
        //  edi height
        height: 26rem;
        
        background: #118899;
        border-radius: 12px;
        .card_title {
          h1 {
            font-family: "Helvetica";
            font-style: normal;
            font-weight: 400;
            font-size: 20px;
            line-height: 24px;
           

            color: #ffffff;
          }
        }
        .cardContainer {
          display: flex;
          justify-content: center;
          align-items: center;
          box-sizing: border-box;
          // min-width: 260px;

          // min-height: 281px;
          background: #f2f8fb;
          border: 1px solid #f2f8fb;
          box-shadow: 0px 6px 12px -2px rgba(0, 0, 0, 0.2);
          border-radius: 12px;
          // height: 22rem;
          width:100%;
          height: 100%;
          p {
            font-family: "Helvetica";
            font-style: normal;
            font-weight: 400;
            font-size: 14px;
            line-height: 24px;
           
            text-align: center;

            color: #323232;
          }
        }
        .nextIcons {
          position: absolute;
          bottom: -0.3rem;
          right: 0;
          width: 68px;
          height: 48px;
        }
      }
    }
  }
}
