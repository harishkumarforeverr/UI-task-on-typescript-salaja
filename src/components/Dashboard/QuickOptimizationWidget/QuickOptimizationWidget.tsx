import React, { useEffect, useState } from "react";

import nextIcon from "../Assests/nextIcon.png";
import "./QuickOptimizationWidget.scss";
import { Button, Card, Checkbox, Select } from "antd";
import SelectWrapper from "../SelectWrapper/SelectWrapper";
import { CheckboxChangeEvent } from "antd/es/checkbox";
import { Assests } from "../Assests";
import { CheckCircleOutlined } from "@ant-design/icons";
function QuickOptimizationWidget({ setView }: { setView: any }) {
  const cardsObj = [
    {
      title: "Quick Spin",
      subTitle: "",
      desc: "Get your motor spining in",
      subDesc: "just few steps",
    },
    {
      title: "Optimization",
      subTitle: "Wizards",
      desc: "Go here after your motor",
      subDesc: "spin consistently",
    },
    {
      title: "Advanced",
      subTitle: "Tuning",
      desc: "Access all your control",
      subDesc: "in one single page",
    },
    {
      title: "Register Map",
      subTitle: "",
      desc: "Full interactive device",
      subDesc: "register map",
    },
  ];
  const [showCards, setShowCards] = useState(false);
  const [setup, setSetup] = useState(false);
  const onChange = (e: CheckboxChangeEvent) => {
    const value = e.target.checked;
    setShowCards(value);
  };
  const handleTheSetup = () => {
    console.log("sssssssssssss");
    setSetup(true);
  };
  return (
    <div className="QuickOptimizationWidget">
      <div className="Conatiner2sIconsTitle">
        <>
          <div className="QuickOptimizationWidget_hardwareSetup">
            <div>
              <SelectWrapper
                BtnSelectDeviceOpacity="0.5"
                dotColor="#40C36C"
                deviceTitle="MCF8316EVM"
                devDesc="This GUI supports MCF8316EVM"
                devSubDesc="Correct device detected : MCF8316A"
              />
            </div>
            <div className="hardwareSetup">
              <div className="hardWareTitleAndImage">
                <h1>Hardware Setup</h1>

                <img src={Assests.DashboardImageCircut} alt="Assests" />
              </div>
              <div className="hardWareDescription">
                <h1>
                  Ensure the hardware is configured according to the below steps
                  :
                </h1>
                <div>
                  <ol>
                    <li>
                      J1 selected to POT, J3 selected to 5V_USB, and J5 selected
                      to 3V3COM
                    </li>
                    <li>Potentiometer R4 is rotated clockwise fully</li>
                    <li>
                      S1 pushed right to select BRAKE=RUN, S2 pushed left to
                      select DIR=ABC, and
                    </li>
                    <li> S3 pushed right to select DRVOFF=ON</li>
                  </ol>
                  <div>
                    <p>
                      Connect the motor to J8. Supply a voltage compliant with
                      the Power Supply Voltage (VM). Recommended voltage range
                      is 4.5V â€“ 35V. Connect the positive terminal to VBAT (PIN3
                      of J7) and negative terminal to PGND (PIN2 of J7).Plug in
                      the micro-USB, from the computer that holds the GUI, to
                      the USB Connector J2.
                    </p>
                  </div>
                  {setup ? (
                    <div className="setupdone">
                      <h1>
                        <span className="icon">
                          <CheckCircleOutlined />
                        </span>
                        The Hardware setup is done
                      </h1>
                    </div>
                  ) : (
                    <>
                      <div className="actionBtn">
                        <Checkbox onChange={onChange} />
                        <p>Hardware setup has been done manually</p>
                        {showCards ? (
                          <Button
                            className="Setup_completed_btn"
                            onClick={handleTheSetup}
                          >
                            Setup completed
                          </Button>
                        ) : (
                          <Button>Setup Now</Button>
                        )}
                      </div>
                    </>
                  )}
                </div>
              </div>
            </div>
          </div>

          <div
            style={{
              opacity: setup ? 1 : 0.4,
              cursor: !setup ? "not-allowed" : "pointer",
            }}
            className="cardsConatiner"
          >
            {cardsObj.map((obj) => {
              return (
                <div key={obj.title} className="cardsConatiner_card">
                  <div className="cards">
                    <div className="card_title">
                      <h1>
                        {obj.title}

                        {obj.subTitle}
                      </h1>
                    </div>
                    <div className="cardContainer">
                      {" "}
                      <div>
                        <p> {obj.desc} </p>
                        <p>{obj.subDesc}</p>
                      </div>
                    </div>
                    <img
                      className="nextIcons"
                      src={!setup ? nextIcon : Assests.nextIcon_pro}
                      alt="ok"
                    />
                  </div>
                </div>
              );
            })}
          </div>
          {/* )} */}
        </>
      </div>
    </div>
  );
}

export default QuickOptimizationWidget;
