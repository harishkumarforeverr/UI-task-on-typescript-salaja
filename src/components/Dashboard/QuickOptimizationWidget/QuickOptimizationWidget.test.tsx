import { render, screen } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import renderer from "react-test-renderer"; 
import QuickOptimizationWidget from "./QuickOptimizationWidget";

const mockHistoryPush = jest.fn();
const mockGoBack = jest.fn();

jest.mock("react-redux", () => ({
  ...jest.requireActual("react-redux"),
  useSelector: jest.fn(),
  useDispatch: jest.fn(),
}));

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useLocation: () => ({
    pathname: "/",
  }),
  useHistory: () => ({
    push: mockHistoryPush,
    goBack: mockGoBack,
  }),
  useParams: () => ({
    pathParam: "haribusine-HOAoFA",
  }),
  useNavigate: () => ({
    pathParam: "haribusine-HOAoFA",
  }), 
}));
const setView=jest.fn((arg:any)=>arg)
test('renders the compoenet and finding whether a text present in the dom or not', () => {
  const MyQuickOptimizationWidget=()=>{
    return (
      <Router>
        <QuickOptimizationWidget setView={setView} />
      </Router>
    )
  }
  render(<MyQuickOptimizationWidget />);
  const linkElement = screen.getByText("in one single page");
  expect(linkElement).toBeInTheDocument();
});

 test('renders the and finding whether a text present in the dom or not', () => {
  const MyQuickOptimizationWidget=()=>{
    return (
      <Router>
        <QuickOptimizationWidget setView={setView} />
      </Router>
    )
  }
  render(<MyQuickOptimizationWidget />);
  const linkElement = screen.getByText("Hardware Setup");
  expect(linkElement).toBeInTheDocument();
}); 

it("creating the snapshot of the compoenent correctly", () => {
  const MyQuickOptimizationWidget = () => {
    return (
      <Router>
        <QuickOptimizationWidget setView={setView} />
      </Router>
    );
  };
  const tree = renderer.create(<MyQuickOptimizationWidget />).toJSON();
  expect(tree).toMatchSnapshot();
});
