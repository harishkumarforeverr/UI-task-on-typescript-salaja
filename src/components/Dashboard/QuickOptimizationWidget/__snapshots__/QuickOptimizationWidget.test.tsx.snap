// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`creating the snapshot of the compoenent correctly 1`] = `
<div
  className="QuickOptimizationWidget"
>
  <div
    className="Conatiner2sIconsTitle"
  >
    <div
      className="QuickOptimizationWidget_hardwareSetup"
    >
      <div>
        <div
          className="ant-card conatinerCard css-dev-only-do-not-override-1n7nwfa"
          style={
            Object {
              "width": 300,
            }
          }
        >
          <div
            className="ant-card-body"
            style={Object {}}
          >
            <div>
              <div
                className="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
              >
                <label
                  className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink={true}
                  id="selectlabelCommon"
                >
                  Select Device*
                </label>
                <div
                  className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl  css-4gwozz-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
                  onClick={[Function]}
                >
                  <div
                    aria-expanded="false"
                    aria-haspopup="listbox"
                    aria-labelledby="selectlabelCommon demo-simple-select"
                    className="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                    id="demo-simple-select"
                    onBlur={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onMouseDown={[Function]}
                    role="button"
                    tabIndex={0}
                  >
                    Select Your Device
                  </div>
                  <input
                    aria-hidden={true}
                    className="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                    disabled={false}
                    onAnimationStart={[Function]}
                    onChange={[Function]}
                    required={false}
                    tabIndex={-1}
                    value="Select Your Device"
                  />
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
                    data-testid="ArrowDropDownIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M7 10l5 5 5-5z"
                    />
                  </svg>
                  <fieldset
                    aria-hidden={true}
                    className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      className="css-14lo706"
                    >
                      <span>
                        Select Device*
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
            <div>
              <div
                className="deatiles"
              >
                <span
                  className="ColorContainer"
                  style={
                    Object {
                      "background": "#40C36C",
                    }
                  }
                />
                <span
                  className="numberText"
                >
                  MCF8316EVM
                </span>
              </div>
              <p
                className="GuiSupport"
              >
                This GUI supports MCF8316EVM
              </p>
              <p
                className="GuiSupport"
                style={
                  Object {
                    "fontSize": "12px",
                    "opacity": "0.8",
                  }
                }
              >
                Correct device detected : MCF8316A
              </p>
            </div>
            <button
              className="ant-btn css-dev-only-do-not-override-1n7nwfa ant-btn-default lightRed btn"
              disabled={false}
              onClick={[Function]}
              style={
                Object {
                  "opacity": "0.5",
                }
              }
              type="button"
            >
              <span>
                Select Device
              </span>
            </button>
          </div>
        </div>
      </div>
      <div
        className="hardwareSetup"
      >
        <div
          className="hardWareTitleAndImage"
        >
          <h1>
            Hardware Setup
          </h1>
          <img
            alt="Assests"
            src="DashboardImageCircut.png"
          />
        </div>
        <div
          className="hardWareDescription"
        >
          <h1>
            Ensure the hardware is configured according to the below steps :
          </h1>
          <div>
            <ol>
              <li>
                J1 selected to POT, J3 selected to 5V_USB, and J5 selected to 3V3COM
              </li>
              <li>
                Potentiometer R4 is rotated clockwise fully
              </li>
              <li>
                S1 pushed right to select BRAKE=RUN, S2 pushed left to select DIR=ABC, and
              </li>
              <li>
                 S3 pushed right to select DRVOFF=ON
              </li>
            </ol>
            <div>
              <p>
                Connect the motor to J8. Supply a voltage compliant with the Power Supply Voltage (VM). Recommended voltage range is 4.5V â€“ 35V. Connect the positive terminal to VBAT (PIN3 of J7) and negative terminal to PGND (PIN2 of J7).Plug in the micro-USB, from the computer that holds the GUI, to the USB Connector J2.
              </p>
            </div>
            <div
              className="actionBtn"
            >
              <label
                className="ant-checkbox-wrapper css-dev-only-do-not-override-1n7nwfa"
              >
                <span
                  className="ant-checkbox css-dev-only-do-not-override-1n7nwfa"
                  style={Object {}}
                >
                  <input
                    checked={false}
                    className="ant-checkbox-input"
                    disabled={false}
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyPress={[Function]}
                    onKeyUp={[Function]}
                    type="checkbox"
                  />
                  <span
                    className="ant-checkbox-inner"
                  />
                </span>
              </label>
              <p>
                Hardware setup has been done manually
              </p>
              <button
                className="ant-btn css-dev-only-do-not-override-1n7nwfa ant-btn-default"
                disabled={false}
                onClick={[Function]}
                type="button"
              >
                <span>
                  Setup Now
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="cardsConatiner"
      style={
        Object {
          "cursor": "not-allowed",
          "opacity": 0.4,
        }
      }
    >
      <div
        className="cardsConatiner_card"
      >
        <div
          className="cards"
        >
          <div
            className="card_title"
          >
            <h1>
              Quick Spin
            </h1>
          </div>
          <div
            className="cardContainer"
          >
             
            <div>
              <p>
                 
                Get your motor spining in
                 
              </p>
              <p>
                just few steps
              </p>
            </div>
          </div>
          <img
            alt="ok"
            className="nextIcons"
            src="nextIcon.png"
          />
        </div>
      </div>
      <div
        className="cardsConatiner_card"
      >
        <div
          className="cards"
        >
          <div
            className="card_title"
          >
            <h1>
              Optimization
              Wizards
            </h1>
          </div>
          <div
            className="cardContainer"
          >
             
            <div>
              <p>
                 
                Go here after your motor
                 
              </p>
              <p>
                spin consistently
              </p>
            </div>
          </div>
          <img
            alt="ok"
            className="nextIcons"
            src="nextIcon.png"
          />
        </div>
      </div>
      <div
        className="cardsConatiner_card"
      >
        <div
          className="cards"
        >
          <div
            className="card_title"
          >
            <h1>
              Advanced
              Tuning
            </h1>
          </div>
          <div
            className="cardContainer"
          >
             
            <div>
              <p>
                 
                Access all your control
                 
              </p>
              <p>
                in one single page
              </p>
            </div>
          </div>
          <img
            alt="ok"
            className="nextIcons"
            src="nextIcon.png"
          />
        </div>
      </div>
      <div
        className="cardsConatiner_card"
      >
        <div
          className="cards"
        >
          <div
            className="card_title"
          >
            <h1>
              Register Map
            </h1>
          </div>
          <div
            className="cardContainer"
          >
             
            <div>
              <p>
                 
                Full interactive device
                 
              </p>
              <p>
                register map
              </p>
            </div>
          </div>
          <img
            alt="ok"
            className="nextIcons"
            src="nextIcon.png"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
